src/sentry/snuba/metrics/query_builder.py

```diff
@@ -89,6 +89,25 @@
 QUERY_PROJECT_LIMIT = 10
 
 
+def _strip_project_id(condition):
+    if isinstance(condition, BooleanCondition):
+        new_boolean_condition = BooleanCondition()
+        for nested_condition in condition.conditions:
+            stripped_condition = _strip_project_id(nested_condition)
+            # Only if we have a condition object we append, since if its None it means that we directly had a
+            # project.id.
+            if stripped_condition is not None:
+                new_boolean_condition.conditions.append(stripped_condition)
+
+        return new_boolean_condition
```

Comment:
My suggestion had a small problem, if a `BooleanCondition` has no valid sub-conditions, you will return an empty object. To solve this you could add the following:

```suggestion
        return new_boolean_condition if new_boolean_condition.conditions else None
```